### PR TITLE
Improve test coverage on opensearch controller

### PR DESCRIPTION
**Pull request by ddrmanxbxfr**
Hi guys,

I've found that the OpenSearch controller wasn't unit tested at all.

Here's a controller test for it!

Thanks!

### CODE REVIEW

File: test/controllers/opensearch_controller_test.exs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+defmodule HexWeb.OpensearchControllerTest do
+  use HexWeb.ConnCase, async: true
+
+  test "opensearch" do
+    conn = get build_conn(), "/hexsearch.xml"
+
+    path          = Path.join([__DIR__, "..", "fixtures"])
+    expected_body = File.read!(Path.join(path, "opensearch.xml"))
+
+    assert response(conn, 200) == expected_body
```
Comment (ericmj) [inline test/controllers/opensearch_controller_test.exs]: We don't need to test the exact contents. You can use `=~` to match against a subset.

Comment (ddrmanxbxfr) [inline test/controllers/opensearch_controller_test.exs]: Indeed, fixed.

Thanks!

File: test/controllers/opensearch_controller_test.exs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+defmodule HexWeb.OpensearchControllerTest do
+  use HexWeb.ConnCase, async: true
+
+  test "opensearch" do
+    conn = get build_conn(), "/hexsearch.xml"
+
+    path          = Path.join([__DIR__, "..", "fixtures"])
+    expected_body = File.read!(Path.join(path, "opensearch.xml"))
+
+    assert response(conn, 200) =~ expected_body
```
Comment (ericmj) [inline test/controllers/opensearch_controller_test.exs]: We should only match against a subset. There is no need to include a fixture that is an exact copy of the expected output. We can match against something like: `<Url type="text/html" method="get" template="http://localhost:4001/packages?search={searchTerms}&amp;sort=downloads" />`.

File: test/controllers/opensearch_controller_test.exs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+defmodule HexWeb.OpensearchControllerTest do
```
Comment (wojtekmach) [inline test/controllers/opensearch_controller_test.exs]: We've recently changed project structure [1]; please change this to `Hexpm.Web.OpenSearchControllerTest` and update filename too.

**Comment by ericmj** [comment]
Ping? :)

**Comment by ddrmanxbxfr** [comment]
:D I'm giving it a go tonight. Thanks for the recall.

File: test/controllers/opensearch_controller_test.exs (hunk 3 above)
Comment (ddrmanxbxfr) [inline test/controllers/opensearch_controller_test.exs]: Done. What would be the new filename ?

File: test/controllers/opensearch_controller_test.exs (hunk 2 above)
Comment (ddrmanxbxfr) [inline test/controllers/opensearch_controller_test.exs]: Indeed, added the matcher instead !

**Comment by ddrmanxbxfr** [comment]
Moved the files and updated the test with a matcher instead of fixtures.

**Pull request merged**